Apps/Inc/Tasks.h

```diff
@@ -157,12 +152,27 @@ typedef enum{
 extern fault_bitmap_t FaultBitmap;
 extern os_error_loc_t OSErrLocBitmap;
 
+/**
+ * Fault Exception Struct
+*/
+typedef struct{
+    uint8_t prio;
+    const char* message; 
+    volatile void (*callback)();
```

Comment:
I think we declared the callback as volatile because we thought other threads might change FaultState's exception variable, but I suppose since CurrException actually gets set in the same FaultState.c it isn't necessary?

Reply:
As for the exception_t error, sophistication sounds nice, but we're currently not sure how we would use this or what changes to make.